File: server/src/schemas/filters.py
from pydantic import BaseModel, Field
from typing import Literal


CategoryKey = Literal["swim", "bike", "run", "electronics", "slots"]

class SubcategoryItem(BaseModel):
    key: str
    label: str
    requires_size: bool = False


class SubcategoryGroup(BaseModel):
    name: str
    items: list[SubcategoryItem]


class CategoryFilter(BaseModel):
    key: Literal["swim", "bike", "run", "electronics", "slots"]
    label: str
    groups: list[SubcategoryGroup] | None = None
    items: list[SubcategoryItem] | None = None
    default_tags: list[str] = Field(default_factory=list)


class GeoItem(BaseModel):
    key: str
    name: str
    flag: str | None = None
    cities: list[str] = Field(default_factory=list)


class FilterConfig(BaseModel):
    categories: list[CategoryFilter]
    countries: list[GeoItem]
    default_cities: list[str]
    conditions: list[dict[str, str]]
    sizes: list[str]
    ad_types: list[dict[str, str]]

File: client/src/types/filters.ts
export interface SubcategoryItem {
  key: string;
  label: string;
  requires_size: boolean;
}

export interface SubcategoryGroup {
  name: string;
  items: SubcategoryItem[];
}

export interface CategoryFilter {
  key: string;
  label: string;
  groups: SubcategoryGroup[] | null;
  items: SubcategoryItem[] | null;
  default_tags: string[];
}

export interface GeoItem {
  key: string;
  name: string;
  flag: string | null;
  cities: string[];
}

export interface FilterConfig {
  categories: CategoryFilter[];
  countries: GeoItem[];
  default_cities: string[];
  conditions: { key: string; label: string }[];
  sizes: string[];
  ad_types: { key: string; label: string }[];
}

export interface ProductFilters {
  category?: string;
  subcategory?: string;
  country?: string;
  city?: string;
  condition?: string;
  ad_type?: string;
  min_price?: number;
  max_price?: number;
  page?: number;
  limit?: number;
}